libs/matdbg/src/DebugServer.cpp

```diff
@@ -193,8 +193,17 @@ DebugServer::addMaterial(const CString& name, const void* data, size_t size, voi
         return {};
     }
 
-    const uint32_t seed = 42;
-    const MaterialKey key = utils::hash::murmurSlow((const uint8_t*) data, size, seed);
+    // Note that it's possible to have two materials with the exact same content (however wasteful),
+    // but they refer to different instantiation of FMaterial. For that case, we need to make sure
+    // that the key for the hash is different by relying on the user provided pointer. Hence we run
+    // the hash function twice.
+    constexpr uint32_t seed = 42;
+    uint64_t appendedData[2] = {
+        utils::hash::murmurSlow((const uint8_t*) data, size, seed),
+        (uint64_t) userdata,
```

Comment:
Would it be sufficient to only hash the userdata pointer here?

Reply:
good point, but it seems `murmur` hashes to a fix value for <= 8 bytes. I used a 16 bytes value instead.